lib/languageclient.ts

```diff
@@ -91,16 +92,32 @@ export class LanguageClientConnection extends EventEmitter {
   }
 
   /**
-   * Public: Register a callback for a custom message.
+   * Public: Register a callback for a custom notification
    *
    * @param method A string containing the name of the message to listen for.
    * @param callback The function to be called when the message is received.
    *   The payload from the message is passed to the function.
    */
-  public onCustom(method: string, callback: (obj: object) => void): void {
+  public onCustomNotification(method: string, callback: (obj: object) => void): void {
     this._onNotification({ method }, callback)
   }
 
+  // @deprecated Use onCustomNotification method instead
+  public onCustom(method: string, callback: (obj: object) => void): void {
+    this.onCustomNotification(method, callback)
+  }
```

Comment:
I'm not sure this is the right way to deprecate in TypeScript and probably it's too harsh to deprecate it at all. I just think `onCustom` name is confusing, because it's only about notifications.

Reply:
I think it requires a doc comment.

```suggestion
  /**
   * @deprecated Use `onCustomNotification` method instead
   */
  public onCustom(method: string, callback: (obj: object) => void): void {
    this.onCustomNotification(method, callback)
  }
```